src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala

```diff
@@ -238,6 +225,29 @@ trait PodStatusConversion {
           PodInstanceState.Stable -> None
     }
   }
+
+  /**
+    * @return the resources actually allocated/in-use by this pod instance; terminal instances don't count
+    */
+  def allocatedResources(pod: PodDefinition, instance: Instance): Option[Resources] = {
+    import InstanceStatus._
+
+    instance.state.status match {
+      case Staging | Starting | Running | Reserved | Unreachable | Killing =>
```

Comment:
There are several instance states, that can differ from task states: e.g the instance will go into state Failed, if one task fails. The whole instance still would consume resources.

A high coupling is reflected here between the instance state and the task state, that we probably should avoid.

2 options here:
1) you return always `Resources` where the actual values are 0
2) you maintain an `Option[Resources]`, if resources are 0